cairo/ethereum_types/bytes.cairo

```diff
@@ -50,6 +73,17 @@ struct TupleBytes {
     value: TupleBytesStruct*,
 }
 
+// Important note about dictionary/mapping types:
+// Since keys are stored as pointers, accessing a dictionary with equivalent but distinct
+// key objects may not work as expected. For example:
+//
+// If dict[b'123'] = b'345' is set, accessing with k = b'123' later may not find the value
+// since k points to a different memory location than the original key, even though the
+// content is identical.
+//
+// This could benefit from a redesign of the dictionary/mapping types to internally
+// use the hash of the key instead of the key pointer.
```

Comment:
we probably need to maintain two dict for ones: one hash->key and another hash->value

Reply:
to update once done